### PR TITLE
Rchain 3509 remove source

### PR DESCRIPTION
**Pull request by kayvank**
## Overview
<sup>_Remove source of produce/consume from rspace datum & continuation to reduce storage foot-print_</sup>

### JIRA ticket:
<sup>_https://rchain.atlassian.net/browse/RCHAIN-3509_</sup>

### Notes
<sup>_Please check jira ticket for detail analysis of the rspace metrics_</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by kayvank** [comment]
bors try

File: rspace/src/main/scala/coop/rchain/rspace/SpaceMatcher.scala [hunk 1]
```diff
@@ -92,36 +90,35 @@ private[rspace] trait SpaceMatcher[F[_], C, P, A, K] extends ISpace[F, C, P, A,
...
-                                           None +: acc
-                                         )
-                                     }
-                  } yield dataCandidates
-                case _ => acc.reverse.pure[F]
-              }
-      } yield res
-    }
+    for {
+      _ <- spanF.mark(extractSpanLabel)
```
Comment (dzajkowski) [inline rspace/src/main/scala/coop/rchain/rspace/SpaceMatcher.scala:94]: change `trace` => `mark` is outside of scope

File: rspace/src/main/scala/coop/rchain/rspace/history/HistoryInstances.scala [hunk 2]
```diff
@@ -66,22 +66,23 @@ object HistoryInstances {
         case TriePath(Vector(), None, Nil) => // kickstart a trie
```
Comment (dzajkowski) [inline rspace/src/main/scala/coop/rchain/rspace/history/HistoryInstances.scala:66]: this change is also outside of scope

File: rspace/src/main/scala/coop/rchain/rspace/history/HistoryRepositoryImpl.scala [hunk 3]
```diff
@@ -148,8 +148,8 @@ final case class HistoryRepositoryImpl[F[_]: Sync, C, P, A, K](
         s"${key.toHex};delete-join;0"
     }
 
-  private def transform(actions: List[HotStoreAction]): List[Result] =
-    actions.map {
+  private def transform(action: HotStoreAction): Result =
```
Comment (dzajkowski) [inline rspace/src/main/scala/coop/rchain/rspace/history/HistoryRepositoryImpl.scala:151]: here too

**Comment by dzajkowski** [comment]
please rebase on dev

File: rspace/src/main/scala/coop/rchain/rspace/history/HistoryRepositoryImpl.scala (hunk 3 above)
Comment (kayvank) [reply rspace/src/main/scala/coop/rchain/rspace/history/HistoryRepositoryImpl.scala:151]: looks like I didnt do a rebase. the change was introduced with the commit: 3a987cfa

**Pull request closed without merging**